### ISSUE TITLE
How training works ?

### ISSUE THREAD

**Issue by ariandemnika** [huggingface/diffusers#1790]
Hi, I've tried different things on training, i tried with text encoder worked good, but the results on each prompt was the same, I'have tried first with text encoder then added more steps without text encoder it was the same, when i tried without text encoder in 1000 steps with 10 images every prompt was really good but the person(face) was not the same as the images, I have tried in google colab with this [REPOSITORY](https://github.com/TheLastBen/fast-stable-diffusion) it worked really good. i dont know what am i missing.

here are my lately training code:

`os.system("/opt/conda/envs/diffusers/bin/python "+directory+"diffusers/examples/dreambooth/train_dreambooth.py --pretrained_model_name_or_path=\""+directory+"stable-diffusion-v1-5\" --instance_data_dir="+directory+""+q_id+"_cropped"" --class_data_dir=\""+directory+"class/"+gender+"\" --output_dir=\""+directory+""+q_id+"_output\" --train_text_encoder --with_prior_preservation --prior_loss_weight=1.0 --instance_prompt=\"a photo of "+q_id+" person\" --class_prompt=\"a photo of person\" --resolution=512 --train_batch_size=1 --save_steps=100000 --use_8bit_adam --gradient_checkpointing --learning_rate=2e-6 --lr_scheduler=\"constant\" --lr_warmup_steps=0 --max_train_steps=1800")`

Can you please give me some suggestions.

**Comment by averad** [comment]
Did you follow https://github.com/Excalibro1/fast-stable-diffusionwik/wiki/fast-stable-diffusion-wiki

**Comment by ariandemnika** [comment]
> Did you follow https://github.com/Excalibro1/fast-stable-diffusionwik/wiki/fast-stable-diffusion-wiki
> 
> Did you check https://github.com/TheLastBen/fast-stable-diffusion/discussions

Yes on google colab everything works well for me because it first trains the text encoder then the UNET, but some of the arguments which are used in google colab i cant use in AWS EC2 Instance because they don't exist!

My results are good if i use 100+ steps on generating images but i think im missing something it should work good with 30steps, also im using CodeFormer for high resolution images.

**Comment by ariandemnika** [comment]
hi @patrickvonplaten can you check this out, how can i do it on EC2 AWS like fastben did on google colab, he uses some argumets that it doesnt allow me to use on ec2

**Comment by averad** [comment]
## Formatted Training Code Referenced in Initial Post

```
os.system("/opt/conda/envs/diffusers/bin/python "+directory+"diffusers/examples/dreambooth/train_dreambooth.py 
--pretrained_model_name_or_path=\""+directory+"stable-diffusion-v1-5\"
--instance_data_dir="+directory+""+q_id+"_cropped""
--class_data_dir=\""+directory+"class/"+gender+"\" 
--output_dir=\""+directory+""+q_id+"_output\"
--train_text_encoder --with_prior_preservation
--prior_loss_weight=1.0 --instance_prompt=\"a photo of "+q_id+" person\"
--class_prompt=\"a photo of person\"
--resolution=512 --train_batch_size=1 
--save_steps=100000 --use_8bit_adam 
--gradient_checkpointing
--learning_rate=2e-6
--lr_scheduler=\"constant\"
--lr_warmup_steps=0
--max_train_steps=1800")
```

**Comment by averad** [comment]
@ariandemnika thanks for the clarification of the reported issue. Your unable to duplicate results you received using Automatics1111 + Dreambooth when using https://huggingface.co/docs/diffusers/training/dreambooth

I believe your missing `--num_class_images=200 `

**Comment by ariandemnika** [comment]
> @ariandemnika thanks for the clarification of the reported issue. Your unable to duplicate results you received using Automatics1111 + Dreambooth when using https://huggingface.co/docs/diffusers/training/dreambooth `with_prior_preservation`
> 
> I believe your missing `--num_class_images=200 `
> 
> > it’s recommended to generate num_epochs * num_samples images for prior preservation. 200-300 works well for most cases.
> 
> https://huggingface.co/docs/diffusers/training/dreambooth#training-with-a-priorpreserving-loss

@averad thank you for the reply, how could i've set the `num_epochs * num_samples` ?
 I've also tried with `--num_class_images=200 ` forgot to mention that sorry.

**Comment by patrickvonplaten** [comment]
Hey @ariandemnika,

Note that if https://github.com/TheLastBen/fast-stable-diffusion works well for you I'd just recommend using it - it's exactly the same as our example: https://github.com/huggingface/diffusers/tree/main/examples/dreambooth.

I'm not really sure how to help here, could you maybe show your error in a google colab?
